14-deny-external-egress-traffic.md

```diff
@@ -32,8 +32,10 @@ spec:
       protocol: UDP
     - port: 53
       protocol: TCP
-  - to:
-    - namespaceSelector: {}
+    to:
+    - namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: kube-system
```

Comment:
This assumes the kube-system namespace is always labeled which i'm not sure is always true. I think without the namespaceSelector we can at least ensure the policy will work on any k8s implementation

Reply:
AFAIK it is always labeled, according to https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-metadata-name:

> *kubernetes.io/metadata.name*
>
> Example: kubernetes.io/metadata.name: "mynamespace"
>
> Used on: Namespaces
>
> The Kubernetes API server (part of the [control plane](https://kubernetes.io/docs/reference/glossary/?all=true#term-control-plane)) sets this label on all namespaces. The label value is set to the name of the namespace. You can't change this label's value.
>
> This is useful if you want to target a specific namespace with a label [selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/).
kubernetes.io/limit-ranger


So I think this is save to apply.

Reply:
I'm a bit confused with this example as I thought this specification also only allows internal traffic via port 53, so it forbids access to port 80 on cluster-internal pods, which means it would also prevent internal traffic. Isn't this the case here, too ? (regardless of whether the namespaceSelector selects every namespace or only a single one).

Reply:
So the label metadata.name on namespaces is new since 1.21 https://github.com/kubernetes/website/commit/6c935b6f9098b79e9df996b98da34cb56d53046f

So you are correct it's safe to assume the label exist